test/unit/config/set_test.js

```diff
@@ -0,0 +1,85 @@
+goog.module('measurementLibrary.testing.config.configProcessors.registerEventAndSet');
+goog.setTestOnly();
+const {registerEventAndSet_} = goog.require('measurementLibrary.config.configProcessors');
+const DataLayerHelper = goog.require('dataLayerHelper.helper.DataLayerHelper');
+
+describe('the behavior after a call to save', () => {
+  let measure;
+  let helper;
+  let processor;
+  let storage;
+
+  beforeEach(() => {
+    const dataLayer = [];
+    measure = function() {
+      dataLayer.push(arguments);
+    };
+    helper = new DataLayerHelper(dataLayer);
+    processor = jasmine.createSpyObj('processor',
+        ['persistTime', 'processEvent']);
+    storage = jasmine.createSpyObj('storage', ['load', 'save']);
+    registerEventAndSet_(helper, processor, {}, storage);
+  });
+
+  it('calls persistTime once with parameters key, value', () => {
+    measure('set', 'a key', 3);
+
+    expect(processor.persistTime).toHaveBeenCalledTimes(1);
+    expect(processor.persistTime).toHaveBeenCalledWith('a key', 3);
+  });
+
+  it('calls save on the storage interface if persistTime returns' +
+      'a positive integer', () => {
+    processor.persistTime.and.returnValue(1337);
+    measure('set', 'key', 'value');
+
+    expect(storage.save).toHaveBeenCalledTimes(1);
+    expect(storage.save).toHaveBeenCalledWith('key', 'value', 1337);
+  });
+
+  it('calls save on the storage interface if persistTime returns ' +
+      'infinity.',
+      () => {
+        processor.persistTime.and.returnValue(
```

Comment:
Is it possible persistTime could return null/undefined? If so, I'd add a test for it (even just as documentation).

Reply:
Returning undefined would violate the contract.